### PR TITLE
213 mobile internationalization

### LINKED ISSUE #213: [MOBILE] Internationalization
- Handle the translation of the app (l10n)
- Be able to change the language of the app

### PR DESCRIPTION
**Pull request by tbellicha**
Translation of the app: FR (default), EN
Save the language on the device
Able to change the language in the settings
Format of some files
Remove unused functions/files
Updates of UT

Coverage: 70.0%

[Screen_recording_20240206_081415.webm](https://github.com/H-chauvet/RISU/assets/72006230/8f595d28-2140-4e90-90f9-e5ec03558198)

Close #213

### CODE REVIEW

File: Application/mobile/app_code/lib/l10n/app_en.arb [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  "accountCreate": "Create an account",
+  "accountCreatedSuccessfully": "Account created successfully.",
+  "accountCreation": "Account creation",
+  "accountDelete": "Delete my account",
+  "accountDeleted": "Account deleted",
+  "accountEmailConfirmationSent": "An email has been sent to {email} to confirm your account.",
+  "accountHasBeenDeleted": "Your account has been deleted.",
+  "actually": "Currently",
+  "add": "Add",
+  "all": "Tous",
```
Comment (31Nathan) [inline Application/mobile/app_code/lib/l10n/app_en.arb:13]: French in en files ?

File: Application/mobile/app_code/lib/l10n/app_fr.arb [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  "accountEmailConfirmationSent": "Un email a été envoyé à {email} pour confirmer votre compte.",
+  "accountHasBeenDeleted": "Votre compte a été supprimé.",
+  "actually": "Actuellement",
+  "add": "Ajouter",
+  "all": "Tous",
+  "allE": "Toutes",
+  "alreadyHaveAnAccount": "Vous avez déjà un compte",
+  "anonymous": "Anonyme",
+  "article": "Article",
+  "articleAlreadyReturned": "La location a déjà été retournée.",
```
Comment (31Nathan) [inline Application/mobile/app_code/lib/l10n/app_fr.arb]: More "rendue" than "retournée"

File: Application/mobile/app_code/lib/l10n/app_fr.arb [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  "articlesDisplayList": "Afficher la liste des articles",
+  "articlesList": "Liste des articles",
+  "articlesListEmpty": "Aucun article disponible.",
+  "availabilityOfAFavoriteArticle": "Disponibilité d'un article favoris",
```
Comment (31Nathan) [inline Application/mobile/app_code/lib/l10n/app_fr.arb:22]: Un article favori

File: Application/mobile/app_code/lib/l10n/app_fr.arb [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  "errorOccurredDuringProfileConfiguration": "Une erreur est survenue lors de la configuration du profil.",
+  "errorOccurredDuringRentReturning": "Une erreur est survenue lors du retour de la location.",
+  "errorOccurredDuringRenting": "Une erreur est survenue lors de la location.",
+  "errorOccurredDuringSavingData": "Une erreur est survenue lors de la sauvegarde des données.",
+  "errorOccurredDuringSavingReview": "Une erreur est survenue lors de la sauvegarde de l'avis.",
+  "errorOccurredDuringSendingMessage": "Une erreur est survenue lors de l'envoi du message.",
+  "errorOccurredDuringSettingProfile": "Une erreur est survenue lors de la configuration du profil.",
+  "errorOccurredDuringSettingStripe": "Une erreur est survenue lors de la configuration de Stripe.",
+  "errorOccurredDuringUpdateUserInformation": "Une erreur est survenue lors de la mise à jour des informations de l'utilisateur.",
+  "fieldRequired": "Ce champ est requis.",
```
Comment (31Nathan) [inline Application/mobile/app_code/lib/l10n/app_fr.arb:66]: requit

**Review by H-chauvet — APPROVED** [review]
Nice job, nothing to say

**Pull request merged**